src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -111,9 +154,18 @@ public InetSocketAddress next(long spinDelay) {
             lastIndex = 0;
         }
 
-        return serverAddresses.get(currentIndex);
+        InetSocketAddress curAddr = serverAddresses.get(currentIndex);
+
+        String curHostString = getHostString(curAddr);
+        List<InetAddress> resolvedAddresses = new ArrayList<InetAddress>(Arrays.asList(this.resolver.getAllByName(curHostString)));
+        if (resolvedAddresses.isEmpty()) {
+            throw new UnknownHostException("No IP address returned for address: " + curHostString);
+        }
+        Collections.shuffle(resolvedAddresses);
```

Comment:
It doesn't't hurt, but do we really need to shuffle the resolved addresses?

Reply:
Correctly configured DNS servers return multiple IP addresses in round robin fashion for a single hostname, so shuffle is not needed. However there was a concern of badly configured DNS servers might not do this properly and we want to workaround it here.

Reply:
Ok.